include/envoy/network/connection.h

```diff
@@ -249,6 +249,21 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    */
   virtual void setBufferLimits(uint32_t limit) PURE;
 
+  /**
```

Comment:
I think adding these methods directly to the Connection interface would break abstraction. This list may grow even longer if more options are to be supported. I wonder if better approach would be do create a separate interface i.e. Envoy::Network::RuntimeSocketOptions (to distinguish it from the Envoy::Network::Socket::Options which are set through listener configuration, or perhaps the existing Options should be renamed to ListenerOptions or some such). And then return a pointer to this new interface from the Network::Connection object.
What do you think?

Reply:
Thanks! I like the recommendation.
To be clear before making the changes. I will implement a new class `RuntimeSocketOptions` in `network/connection.h` and have a pointer/reference available through Network::Connection.

We could have another PR just for changing Envoy::Network::Socket::Options to Envoy::Network::ListernerOptions

Please correct me if, I have diverged.

Reply:
Yes, this sounds like a good plan.

Reply:
+1 to splitting these into their own interface.  If we're not configuring defaults via source/common/runtime/runtime_impl I'd avoid Runtime in the name to avoid confusion.  Dynamic?  Modifiable?